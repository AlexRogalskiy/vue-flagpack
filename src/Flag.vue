<template>
  <div
    :class="`flag
      size-${size}
      ${dropshadow ? 'dropshadow' : ''}
      ${border ? 'border' : ''}
      ${borderRadius ? 'border-radius' : ''}
      ${gradient}`"
    :style="{borderRadius: customBorderRadius}">
    <img :src="`/public/svg/${size}/${code}.svg`">
  </div>
</template>

<script>
export default {
  name: 'Flag',
  props: {
    size: {
      type: String,
      default: 'l',
    },
    code: {
      type: String,
      required: true,
    },
    dropshadow: {
      type: Boolean,
    },
    border: {
      type: Boolean,
      default: true
    },
    borderRadius: {
      type: Boolean,
      default: true,
    },
    customBorderRadius: {
      type: String,
    },
    gradient: {
      type: String,
      default: 'top-down',
    },
  },
};
</script>

<style scoped lang="scss">
.flag {
  display: inline-block;
  overflow: hidden;
  position: relative;

  &.size {
    &-s {
      width: 16px;
      height: 12px;

      &.dropshadow {
        box-shadow: 0 0 1px 0.5px rgba(0,0,0,0.10);
      }

      &.border-radius {
        border-radius: 1px;
      }
    }

    &-m {
      &.dropshadow {
        box-shadow: 0 1px 2px 0 rgba(0,0,0,0.10);
      }

      &.border-radius {
        border-radius: 1.5px;
      }
    }
  }

  &::before {
    content: '';
    width: 100%;
    height: 100%;
    position: absolute;
    display: block;
    mix-blend-mode: overlay;
  }

  &.border {
    &::before {
      border: 1px solid rgba(0, 0, 0, .2);
      mix-blend-mode: overlay;
    }
  }

  &.border-radius {
    &::before {
      border-radius: 1px;
    }
  }

  &.top-down {
    &::before {
      background-image: linear-gradient(0deg, rgba(0,0,0,0.30) 2%, rgba(255,255,255,0.70) 100%);
    }
  }

  img {
    display: block;
  }
}
</style>
