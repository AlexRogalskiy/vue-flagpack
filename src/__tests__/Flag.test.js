import { mount } from '@vue/test-utils'
import Flag from '../Flag.vue'

describe('Test different flag sizes', () => {
  test('Should render a small dutch flag', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'NL',
        size: 's'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-s',
      'border-radius',
      'border'
    ])
  })

  test('Should render a medium dutch flag', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'NL',
        size: 'm'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-m',
      'border-radius',
      'border'
    ])
  })

  test('Should render a large dutch flag', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'NL',
        size: 'l'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border'
    ])
  })

  test('Test default size (should be large)', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'NL'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border'
    ])
  })
})

describe('Test the different props', () => {
  test('Should render a large US flag with drop shadow', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'US',
        dropshadow: true
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border',
      'dropshadow'
    ])
  })

  test('Should render a large US flag with a custom classname', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'US',
        className: 'custom-flag-class'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border',
      'custom-flag-class'
    ])
  })

  test('Should render a large Belgium flag with a real-linear gradient', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'BE',
        gradient: 'real-linear'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border',
      'real-linear'
    ])
  })

  test('Should render a large Belgium flag with a top-down gradient', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'BE',
        gradient: 'top-down'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border',
      'top-down'
    ])
  })

  test('Should render a large Belgium flag with a real-circular gradient', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'BE',
        gradient: 'real-circular'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border',
      'real-circular'
    ])
  })

  test('Should render a large Belgium flag without a border.', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'BE',
        hasBorder: false
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius'
    ])
  })

  test('Should render a large Italian flag with a border.', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'IT',
        hasBorder: true
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border'
    ])
  })

  test('Should render a large Italian flag with a 10px border radius.', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'IT',
        customBorderRadius: '10px'
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border-radius',
      'border'
    ])
    expect(component.attributes('style')).toBe("border-radius: 10px;")
  })

  test('Should render a large Italian flag with no border radius.', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'IT',
        hasBorderRadius: false
      },
      sync: false
    })

    expect(component.element).toMatchSnapshot()
    expect(component.classes()).toStrictEqual([
      'flag',
      'size-l',
      'border'
    ])
  })
})

test('Component with no flag code given along.', () => {
  const component = mount(Flag, {
    sync: false
  })

  expect(component.element).toMatchSnapshot()
  expect(component.classes()).toStrictEqual([
    'flag',
    'size-l',
    'border-radius',
    'border'
  ])
})

describe('Tests if image url are correct', () => {
  const IMG_URL = 'https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg'

  test('Japan - large', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'JO',
        size: 'l'
      },
      sync: false
    })

    const image = component.find('img')

    expect(image.attributes('src')).toBe(`${IMG_URL}/l/JO.svg?sanitize=true`)
  })

  test('Japan - medium', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'JO',
        size: 'm'
      },
      sync: false
    })

    const image = component.find('img')

    expect(image.attributes('src')).toBe(`${IMG_URL}/m/JO.svg?sanitize=true`)
  })

  test('Japan - small', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'JO',
        size: 's'
      },
      sync: false
    })

    const image = component.find('img')

    expect(image.attributes('src')).toBe(`${IMG_URL}/s/JO.svg?sanitize=true`)
  })
})

describe('Test giving different types of Iso codes', () => {
  const IMG_URL = 'https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg'

  test('Canada - numeric', () => {
    const component = mount(Flag, {
      propsData: {
        code: '124',
        size: 's'
      },
      sync: false
    })

    const image = component.find('img')

    expect(image.attributes('src')).toBe(`${IMG_URL}/s/CA.svg?sanitize=true`)
  })

  test('Canada - Alpha-3', () => {
    const component = mount(Flag, {
      propsData: {
        code: 'CAN',
        size: 's'
      },
      sync: false
    })

    const image = component.find('img')

    expect(image.attributes('src')).toBe(`${IMG_URL}/s/CA.svg?sanitize=true`)
  })
})
