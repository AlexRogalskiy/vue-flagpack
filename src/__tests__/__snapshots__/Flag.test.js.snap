// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component with no flag code given along. 1`] = `
<div
  class="flag size-m border-radius"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/NL.svg?sanitize=true"
  />
</div>
`;

exports[`Test different flag sizes Should render a large dutch flag 1`] = `
<div
  class="flag size-l border-radius"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/l/NL.svg?sanitize=true"
  />
</div>
`;

exports[`Test different flag sizes Should render a medium dutch flag 1`] = `
<div
  class="flag size-m border-radius"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/NL.svg?sanitize=true"
  />
</div>
`;

exports[`Test different flag sizes Should render a small dutch flag 1`] = `
<div
  class="flag size-s border-radius"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/s/NL.svg?sanitize=true"
  />
</div>
`;

exports[`Test different flag sizes Test default size (should be large) 1`] = `
<div
  class="flag size-m border-radius"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/NL.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large Belgium flag with a real-circular gradient 1`] = `
<div
  class="flag size-m border-radius real-circular"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/BE.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large Belgium flag with a real-linear gradient 1`] = `
<div
  class="flag size-m border-radius real-linear"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/BE.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large Belgium flag with a top-down gradient 1`] = `
<div
  class="flag size-m border-radius top-down"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/BE.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large Belgium flag without a border. 1`] = `
<div
  class="flag size-m border-radius"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/BE.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large Italian flag with a 10px border radius. 1`] = `
<div
  class="flag size-m border-radius"
  style="border-radius: 10px;"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/IT.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large Italian flag with a border. 1`] = `
<div
  class="flag size-m border-radius border"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/IT.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large Italian flag with no border radius. 1`] = `
<div
  class="flag size-m"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/IT.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large US flag with a custom classname 1`] = `
<div
  class="flag size-m border-radius custom-flag-class"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/US.svg?sanitize=true"
  />
</div>
`;

exports[`Test the different props Should render a large US flag with drop shadow 1`] = `
<div
  class="flag size-m border-radius dropshadow"
>
  <img
    src="https://raw.githubusercontent.com/Yummygum/flag-pack-core/master/svg/m/US.svg?sanitize=true"
  />
</div>
`;
